SharedPackages/BrowserServicesKit/Sources/Subscription/Managers/SubscriptionManagerV2.swift

```diff
@@ -417,10 +418,44 @@ public final class DefaultSubscriptionManagerV2: SubscriptionManagerV2 {
 
     var cachedUserEntitlements: [SubscriptionEntitlement] {
```

Comment:
Ok I don't want to overcomplicate things, but could we tag this `@MainActor` and consider to push it - unless it disrupts a lot of existing code, then we can postpone.